go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -165,41 +165,116 @@ func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewa
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
-		if len(rule.ports) > 0 {
-			match = fmt.Sprintf("%s && ( %s )", match, egressGetL4Match(rule.ports))
+// createLogicalRouterPolicy uses the previously generated elements and creates the logical_router_policy
+// for a specific egressFirewallRouter
+func createLogicalRouterPolicy(priority int, match, action, namespace string) error {
+	_, stderr, err := util.RunOVNNbctl("--id=@logical_router_policy", "create", "logical_router_policy",
+		fmt.Sprintf("priority=%d", priority),
+		match, "action="+action, fmt.Sprintf("external-ids:egressFirewall=%s", namespace),
+		"--", "add", "logical_router", types.OVNClusterRouter, "policies", "@logical_router_policy")
+	if err != nil {
+		// TODO: lr-policy-add doesn't support --may-exist, resort to this workaround for now.
+		// Have raised an issue against ovn repository (https://github.com/ovn-org/ovn/issues/49)
+		if !strings.Contains(stderr, "already existed") {
+			return fmt.Errorf("failed to add policy route '%s' to %s "+
+				"stderr: %s, error: %v", match, types.OVNClusterRouter, stderr, err)
 		}
+	}
+	return nil
+}
 
-		match = fmt.Sprintf("%s && %s\"", match, getClusterSubnetsExclusion())
+type matchTarget struct {
+	kind  matchKind
+	value string
+}
 
-		_, stderr, err := util.RunOVNNbctl("--id=@logical_router_policy", "create", "logical_router_policy",
-			fmt.Sprintf("priority=%d", efStartPriority-rule.id),
-			match, "action="+action, fmt.Sprintf("external-ids:egressFirewall=%s", ef.namespace),
-			"--", "add", "logical_router", types.OVNClusterRouter, "policies", "@logical_router_policy")
-		if err != nil {
-			// TODO: lr-policy-add doesn't support --may-exist, resort to this workaround for now.
-			// Have raised an issue against ovn repository (https://github.com/ovn-org/ovn/issues/49)
-			if !strings.Contains(stderr, "already existed") {
-				return fmt.Errorf("failed to add policy route '%s' to %s "+
-					"stderr: %s, error: %v", match, types.OVNClusterRouter, stderr, err)
-			}
+type matchKind int
+
+const (
+	matchKindV4CIDR matchKind = iota
+	matchKindV6CIDR
+	matchKindV4AddressSet
+	matchKindV6AddressSet
+)
+
+func (m *matchTarget) toExpr() string {
+	switch m.kind {
+	case matchKindV4CIDR:
+		return fmt.Sprintf("ip4.dst == %s", m.value)
+	case matchKindV6CIDR:
+		return fmt.Sprintf("ip6.dst == %s", m.value)
+	case matchKindV4AddressSet:
+		if m.value != "" {
+			return fmt.Sprintf("ip4.dst == $%s", m.value)
 		}
+		return ""
+	case matchKindV6AddressSet:
+		if m.value != "" {
+			return fmt.Sprintf("ip6.dst == $%s", m.value)
+		}
+		return ""
 	}
-	return nil
+	panic("invalid matchKind")
+}
+
+// generateMatch generates the "match" section of ACL generation for egressFirewallRules.
+// It is referentially transparent as all the elements have been validated before this function is called
+// sample output:
+// match=\"(ip4.dst == 1.2.3.4/32) && ip4.src == $testv4 && ip4.dst != 10.128.0.0/14\
```

Comment:
because these are on the ovn_router we have to exclude internal traffic, there is no way around this for now. if we do not block it if a cluster admin introduces deny all `0.0.0.0/0` it will deny all internal traffic as well